### ISSUE TITLE
The results do not match those reported in the paper.

### ISSUE THREAD

**Issue by ikechiyo** [duyali2000/SemanticFlowGraph#6]
Hello,
I used the run_commits_\<encoding\>.sh script from the replication package to conduct experiments on AspectJ, but the accuracy was lower than the results reported in Table 2 of the paper. Are the hyperparameters specified in the script the same as those used in the evaluation experiments?
![図2](https://github.com/user-attachments/assets/ed401fe2-4a5b-4284-85c6-e31704ad6d6e)

**Comment by X-Luminous** [comment]
> Hello, I used the run_commits_<encoding>.sh script from the replication package to conduct experiments on AspectJ, but the accuracy was lower than the results reported in Table 2 of the paper. Are the hyperparameters specified in the script the same as those used in the evaluation experiments? ![図2](https://private-user-images.githubusercontent.com/57423904/357325079-ed401fe2-4a5b-4284-85c6-e31704ad6d6e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU5NzA3OTAsIm5iZiI6MTcyNTk3MDQ5MCwicGF0aCI6Ii81NzQyMzkwNC8zNTczMjUwNzktZWQ0MDFmZTItNGE1Yi00Mjg0LTg1YzYtZTMxNzA0YWQ2ZDZlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTEwVDEyMTQ1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThjMWFlOGMwMTJiMjdlYTEyMjUzZTgxN2M3OTI1ZTQ0OTY2M2I2Y2MzOTAwOTc1YjAzMjBiYTEwYzQyYzUzM2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.6stod0Y78rayePoM_orlypIsoZRi-CzzR3VONaaJTls)

I have the same problem. Have you solved your problem? If so, can you tell me how you set the parameters?

**Comment by duyali2000** [comment]
The hyperparameters for the experiments are selected based on the model's performance across different datasets and may vary accordingly. I would recommend testing different hyperparameter configurations to identify the optimal setup for your use case.

Additionally, maybe a small trick we discovered during our experiments is that training at the commit-level tends to converge faster and is less prone to oscillations. You could try initializing the models at the file-level and hunk-level with the weights obtained from the commit-level model training. This trick is effective in accelerating convergence and improving performance in our experiments.

If you have other questions, feel free to email me at duyl@lamda.nju.edu.cn.
